### PR TITLE
fix(shell-api): prevent max listeners warning MONGOSH-919

### PR DESCRIPTION
**Pull request by rose-m**

### CODE REVIEW

File: packages/shell-api/src/interruptor.ts [hunk 1]
```diff
@@ -17,6 +17,12 @@ export class InterruptFlag {
   private interrupted = false;
   private onInterrupt = new EventEmitter();
 
+  constructor() {
+    // since we might have user code in a very high async fashion
+    // we don't know beforehand how many listeners we have to register
+    this.onInterrupt.setMaxListeners(Number.POSITIVE_INFINITY);
```
Comment (addaleax) [inline packages/shell-api/src/interruptor.ts:23]:
```suggestion
    this.onInterrupt.setMaxListeners(Infinity);
```

for consistency?

```
$ git grep -w POSITIVE_INFINITY .|wc -l
0
$ git grep -w Infinity .|wc -l
23
```

File: packages/shell-api/src/shell-internal-state.ts [hunk 2]
```diff
@@ -209,8 +209,8 @@ export default class ShellInternalState {
       contextObject.console = {};
     }
     for (const key of ['log', 'warn', 'info', 'error']) {
-      contextObject.console[key] = async(...args: any[]): Promise<void> => {
-        return await contextObject.print(...args);
+      contextObject.console[key] = (...args: any[]): any => {
```
Comment (addaleax) [inline packages/shell-api/src/shell-internal-state.ts:212]:
```suggestion
      contextObject.console[key] = (...args: any[]): Promise<void> => {
```

would still be okay and accurate, right?

**Review by addaleax — APPROVED** [review]

File: packages/shell-api/src/shell-internal-state.ts (hunk 2 above)
Comment (rose-m) [reply packages/shell-api/src/shell-internal-state.ts:212]: Ah yeah, removed that just as a first while testing...

File: packages/shell-api/src/interruptor.ts (hunk 1 above)
Comment (rose-m) [reply packages/shell-api/src/interruptor.ts:23]: SGTM!

**Pull request merged**